### DISCUSSION TITLE
No access to new repository with active ssh key on profile

### DISCUSSION THREAD

**Discussion by toszlanyi** [community/community#22409] [New to GitHub]
<p>Hey all - just joining this group. I am on Endeavour OS (Arch based) and using github for a while to contribute to existing projects. Using the terminal for all git tasks…</p>
<p>I am now in trouble again since the access to github is only available with access token or ssh key. When i created a new repository I cannot push to it - somehow the ssh key in my github profile is only recognized for existing repositories. I could yet again create another ssh key but there must be a smarter solution. Couldnt find any help yet in documentation or forum so asking here…</p>
<p>What I have done:</p>
<ul>
<li>create new repository on my git site</li>
<li><code>git clone  https://github.com/...</code></li>
<li>changed to directory</li>
<li>made changes</li>
<li><code>git add --all</code></li>
<li><code>git commit -m "Initial commit"</code></li>
<li><code>git push -u origin main</code></li>
</ul>
<p>now it doesn’t ask for the passphrase for the ssh key but instead for git username and password which doesnt work since summer</p>
<pre><code class="lang-auto">$ git push -u origin main
Username for 'https://github.com': toszlanyi
Password for 'https://toszlanyi@github.com': 
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: Authentifizierung fehlgeschlagen für 'https://github.com/xyz-repo'
</code></pre>
<pre><code class="lang-auto">$ ssh -vT git@github.com
OpenSSH_8.8p1, OpenSSL 1.1.1l  24 Aug 2021
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Connecting to github.com [140.82.121.4] port 22.
debug1: Connection established.
...
debug1: Offering public key: /home/osz/.ssh/id_ed25519 ED25519 SHA256: ...
debug1: Server accepts key: /home/osz/.ssh/id_ed25519 ED25519 SHA256: ...
Enter passphrase for key '/home/osz/.ssh/id_ed25519': 
Authenticated to github.com ([140.82.121.4]:22) using "publickey".
Hi toszlanyi! You've successfully authenticated, but GitHub does not provide shell access.
</code></pre>
<pre><code class="lang-auto">$ git remote -v
origin  https://github.com/toszlanyi/xyz-repo (fetch)
origin  https://github.com/toszlanyi/xyz-repo (push)
</code></pre>
<pre><code class="lang-auto">$ cat .git/config 
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = https://github.com/toszlanyi/xyz-repo
        fetch = +refs/heads/*:refs/remotes/origin/*
        pushurl = https://github.com/toszlanyi/xyz-repo
[branch "main"]
        remote = origin
        merge = refs/heads/main
</code></pre>
<p>On my other own or cloned repositories I can push using the ssh key  <code>id_ed25519</code> that is set up correctly in my account settings. Any idea how to push / access the new <code>xyz-repo</code>?<br>
Any help appreciated.</p>

**Comment by toszlanyi** [comment]
<p>Anyone, who can explain what I missed? Thx.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="toszlanyi" data-post="1" data-topic="210327">
<div class="title">
<div class="quote-controls"></div>
 toszlanyi:</div>
<blockquote>
<pre><code class="lang-auto">$ git remote -v
origin  https://github.com/toszlanyi/xyz-repo (fetch)
origin  https://github.com/toszlanyi/xyz-repo (push)
</code></pre>
</blockquote>
</aside>
<p>Your repository is configured with an HTTPS remote URL, not SSH, that’s why it’s not using your SSH key. You can change the URL to SSH like this:</p>
<pre><code class="lang-auto">git remote set-url origin git@github.com:toszlanyi/xyz-repo.git
</code></pre>
<p>For future reference, when you click the green “Code” button on the repository page you can select the protocol you want to use in the box that comes up to get the right URL.</p>

**Comment by toszlanyi** [comment, marked as answer]
<p>Wicked! Thanks for pointing into the right direction. Your line only changes the fetch url though. But with option <code>--push</code> it also changes the other url.</p>
<pre><code class="lang-auto">git remote set-url --push origin git@github.com:toszlanyi/xyz-repo.git
</code></pre>

**Comment by airtower-luna** [comment]
<p>Peculiar, in my experience it changes both <em>unless</em> you use one of the flags. I wonder if behavior changed between versions?</p>
